unlock-app/src/components/creator/CreatorLocks.js

```diff
@@ -16,10 +16,47 @@ export class CreatorLocks extends React.Component {
     const { showForm } = this.props
     this.state = {
       showDashboardForm: !!showForm,
+      editingLocks: {},
     }
     this.toggleForm = this.toggleForm.bind(this)
   }
 
+  getLockElement(lock) {
+    const key = JSON.stringify(lock)
+    const { editingLocks } = this.state
+    if (editingLocks[lock.address]) {
+      return (
+        <CreatorLockForm
+          key={key}
+          createLock={() => {}}
+          hideAction={() => this.stopEditingLock(lock.address)}
+          {...lock}
+        />
+      )
+    }
+    return (
+      <CreatorLock
```

Comment:
If the lock knows of its state the conditional of whether to display the form or the non form version would happen inside `CreatorLock`.